worldedit-core/src/main/java/com/sk89q/worldedit/util/paste/EngineHubPaste.java

```diff
@@ -49,47 +51,49 @@ private PasteTask(String content, PasteMetadata metadata) {
 
         @Override
         public URL call() throws IOException, InterruptedException {
-            URL initialUrl = HttpRequest.url("https://paste.enginehub.org/signed_paste");
-
-            HttpRequest requestBuilder = HttpRequest.get(initialUrl);
-
-            requestBuilder.header("x-paste-meta-from", "EngineHub");
-            if (metadata.name != null) {
-                requestBuilder.header("x-paste-meta-name", metadata.name);
-            }
-            if (metadata.author != null) {
-                requestBuilder.header("x-paste-meta-author", metadata.author);
+            try (HttpClient client = HttpClient.newHttpClient()) {
+                HttpRequest.Builder signRequestBuilder = HttpRequest.newBuilder()
+                        .uri(URI.create("https://paste.enginehub.org/signed_paste_v2"))
+                        .header("x-paste-meta-from", "EngineHub");
+
+                if (metadata.name != null) {
+                    signRequestBuilder = signRequestBuilder.header("x-paste-meta-name", metadata.name);
+                }
+                if (metadata.author != null) {
+                    signRequestBuilder = signRequestBuilder.header("x-paste-meta-author", metadata.author);
+                }
+                if (metadata.extension != null) {
+                    signRequestBuilder = signRequestBuilder.header("x-paste-meta-extension", metadata.extension);
+                }
+
+                SignedPasteResponse response = GSON.fromJson(
+                        client.send(signRequestBuilder.build(), HttpResponse.BodyHandlers.ofString()).body(),
```

Comment:
Previous code required a status code here. If this request fails, will the new code just continue with some (bad?) data? Is this relying on the fact that the json deserialization will fail if the response doesn't have a specific structure? If so, won't it throw a different exception, which doesn't extend IOException and thus will give an ugly result back instead of a converted message?